### DISCUSSION TITLE
[Crashlytics] Out Of Memory, Hangs (ie: SIGKILL terminations)

### DISCUSSION THREAD

**Discussion by naftaly** [firebase/firebase-ios-sdk#12889] [General]
Why is it that Crashlytics doesn't support any type of SIGKILL terminations such as OOMs or Hangs. I understand it can't be caught like normal signals, but we can use heuristics to determine a likely cause. I remember Fabric used to have some sort of OOM detection in it. So Im just curious if there's any ongoing discussion around this.

**Comment by paulb777** [comment]
#12881 was just contributed yesterday

**Reply by naftaly** [reply]
@paulb777 yes, I pushed that, and it only adds support for when Apple is kind enough to gently terminate an app. There's a lot more to do in order to have real support for memory terminations as well as other kinds of watchdog events.  I'm just curious about the reason(s) behind their surprising absence.
